File: backend/app/models/scene.py
"""Scene-level visual novel contracts."""

from pydantic import Field

from app.models.commands import GameCommand
from app.models.common import StrictBaseModel


class SceneBeat(StrictBaseModel):
    """A structured visual novel scene segment."""

    scene_id: str
    scene_display_name: str | None = None
    title: str
    summary: str
    commands: list[GameCommand] = Field(default_factory=list)
    tags: list[str] = Field(default_factory=list)
    chapter: int = Field(..., ge=0)

File: editor/src/types/scene.ts
import type { GameCommand } from "./commands";

export interface SceneBeat {
  scene_id: string;
  scene_display_name?: string | null;
  title: string;
  summary: string;
  commands: GameCommand[];
  tags: string[];
  chapter: number;
  is_ending?: boolean;
  ending_id?: string;
  ending_title?: string;
}
